doc/mid-based-signalling.md

```diff
@@ -0,0 +1,276 @@
+therefore describe them in the answer:
+
+m.call.answer (full mesh)
+```
+{
+    "m.negotiate": {
+        "version": 1,
+        "call_id": "35657a5b793ce",
+        "conf_id": "bbe53499f82e3",
+        "party_id": "678910",
+        "description": {
+            "type": "answer",
+            "sdp": "[...]",
+        }
+    },
+    "m.call.capabilities": {
+        "m.call.transferee": false,
+        "m.call.dtmf": false,
+    },
+    "m.call.describe": {
+        "1": { // transceiver mid 1
+            "media_uuid": "cccc-cccc-cccccc-cccc-cccc",
+            "media_group_uuid": "2345-2345-234567-2345-2345",
+            "purpose": "m.usermedia",
+            "kind": "video"
+        },
+        "2": {
+            "media_uuid": "bbbb-bbbb-bbbbbb-bbbb-bbbb",
+            "media_group_uuid": "1234-1234-123456-1234-1234",
+            "purpose": "m.usermedia",
+            "kind": "audio",
+        },
+    },
+}
+```
+
+A focus, however, will not send any tracks by default and therefore does not include an
+`m.call.describe` bblock. Instead, it includes an `m.track.advertise` block advertising
+what tracks are available for that `conf_id`.
+
+m.call.answer (focus)
+```
+{
+    "m.negotiate": {
+        "version": 1,
+        "call_id": "35657a5b793ce",
+        "conf_id": "bbe53499f82e3",
+        "party_id": "678910",
+        "description": {
+            "type": "answer",
+            "sdp": "[...]",
+        }
+    },
+    "m.call.capabilities": {
+        "m.call.transferee": false,
+        "m.call.dtmf": false,
+    },
+    "m.call.advertise": {
+        "alice:example.org": { // user ID
+            "88888888": { // device ID
+                "2345-2345-234567-2345-2345": [{ // media group uuid
+                    "media_uuid": "aaaa-aaaa-aaaaaa-aaaa-aaaa":
+                    "purpose": "m.usermedia",
+                    "kind": "video",
+                }, {
+                    "media_uuid": "bbbb-bbbb-bbbbbb-bbbb-bbbb":
+                    "purpose": "m.usermedia",
+                    "kind": "audio",
+                },
+            },
+        }
+    },
+}
+```
+
+XXX: How flat vs deep do we want the structure to be here? I've done it quite deep here,
+organised by user ID / device ID / media group UUID, but they could also just be a flat
+list of tracks. It would be more duplication but maybe less effort to read.
+
+The expected behaviour here would be for foci to essentially maintain a structure with all
+tracks being pushed to it. This structure would probably have call IDs as a top level index,
+then look very similar to the structure of the `m.call.advertise` event. It could either keep
+a reference to the transceiver it was receiving media on in the structure itself alongside
+the media UUID, or maintain a separate map of media UUID to transceiver / peer connection such
+that the first structure could be marshalled to JSON and sent to clients as-is.
```

Comment:
I think the SFU do not need to save and maintain the map of media UUID to transceiver / peer connection. The mid's are only needed to identify the tracks inside the SDP. If the connection is negotiated, they are no longer required.

Reply:
Well, it need to know what media UUID corresponds to what it's sending in each case: whether it does that by mapping them to mid or track ID or whatever is up to the implementation - is that what you meant?

Reply:
Yes, exactly, that is what I meant. I only just wanted to pointed out that's MID's quite unstable and only needed to init this process. And we should not based on it in such `whatever implementation`.

Reply:
Understood – hopefully c06d3d4 clarifies this a bit.